### PR TITLE
[RW-5782][risk=no] Puppeteer: Migrate e2e project tslint to eslint

### PR DESCRIPTION
**Pull request by aweng98**
tslint has been deprecated in favor of eslint.

Uninstalled `tslint-*` lib and removed `tslint-*` config files. Installed `prettier`, `eslint-*` lib. Created new rule files. Fixed many eslint warnings and errors. As result, many changed files (I think these files can be safely skipped review unless you're curious).

The real changes to review are:
`.eslintignore`
`.eslintrc.js`
`package.json`
`tsconfig.json`

cmd `yarn compile` output:
Screen Shot 2021-02-22 at 12 09 14 PM

cmd `yarn lint` output:
Screen Shot 2021-02-22 at 12 10 02 PM

Many eslint warnings/errors could be fixed by `--fix`. I'm saving this task in a follow up PR.
New cmd `yarn format` is not executed in this PR. I'm saving this task in a follow up PR.

### CODE REVIEW

File: e2e/.eslintrc.js [hunk 1]
```diff
@@ -10,4 +19,79 @@ module.exports = {
...
+    'valid-jsdoc': 0,
+    'no-var': 2,
+    'no-console': 0,
+    'prefer-const': 2,
+    'comma-dangle': 0,
+    'arrow-parens': 0,
+    'no-trailing-spaces': 2,
+    'eol-last': [2, 'always'],
+    'no-ternary': 'off',
+    'no-duplicate-imports': 1,
```
Comment (aweng98) [inline e2e/.eslintrc.js:86]: Got a lot errors when ran eslint. In order to build project successfully and not break nothing, I set these rules to warning from error (default). I will change one rule at a time to error in separate PRs in future.

File: e2e/README.md [hunk 2]
```diff
@@ -1,106 +1,125 @@
 # All-of-Us Test Automation with Puppeteer in Chrome or Chromium
-Puppeteer `e2e` end-to-end integration tests.  These tests are run in CircleCI
+
```
Comment (aweng98) [inline e2e/README.md:2]: added note about `yarn precommit`. file formatted.

File: e2e/app/component/annotations-sidebar.ts [hunk 3]
```diff
@@ -1,11 +1,11 @@
-import {Page} from 'puppeteer';
+import { Page } from 'puppeteer';
```
Comment (aweng98) [inline e2e/app/component/annotations-sidebar.ts:1]: file formatted.

File: e2e/app/component/concept-domain-card.ts [hunk 4]
```diff
@@ -1,23 +1,22 @@
-import {Page} from 'puppeteer';
+import { Page } from 'puppeteer';
```
Comment (aweng98) [inline e2e/app/component/concept-domain-card.ts:1]: file formatted.

File: e2e/app/element/base-element.ts [hunk 5]
```diff
@@ -1,23 +1,33 @@
-import {ClickOptions, ElementHandle, Page, WaitForSelectorOptions} from 'puppeteer';
+import {
```
Comment (aweng98) [inline e2e/app/element/base-element.ts:1]: file formatted.

File: e2e/jest.config.js [hunk 6]
```diff
@@ -2,49 +2,44 @@ const { defaults } = require('jest-config');
...
-  "verbose": false,
-  "bail": 1,  // Stop running tests after `n` failures
-  "preset": "jest-puppeteer",
-  "testTimeout": 1200000,
-  "testRunner": "jest-circus/runner",
-  "testEnvironment": "<rootDir>/puppeteer-custom-environment.ts",
-  "setupFilesAfterEnv": [
-    "<rootDir>/jest-circus.setup.ts",
-    "<rootDir>/jest.test-setup.ts"
+  verbose: false,
```
Comment (aweng98) [inline e2e/jest.config.js:5]: file formatted.

File: e2e/tests/cohorts/cohorts-ui.spec.ts [hunk 7]
```diff
@@ -1,12 +1,11 @@
 import CohortBuildPage from 'app/page/cohort-build-page';
-import {PhysicalMeasurementsCriteria} from 'app/page/criteria-search-page';
+import { PhysicalMeasurementsCriteria } from 'app/page/criteria-search-page';
```
Comment (aweng98) [inline e2e/tests/cohorts/cohorts-ui.spec.ts:2]: file formatted.

File: e2e/tests/conceptsets/copy-to-workspace.spec.ts [hunk 8]
```diff
@@ -1,14 +1,14 @@
-import {Domain} from 'app/component/concept-domain-card';
+import { Domain } from 'app/component/concept-domain-card';
```
Comment (aweng98) [inline e2e/tests/conceptsets/copy-to-workspace.spec.ts:1]: file formatted.

File: e2e/tests/nightly/registration.spec.ts [hunk 9]
```diff
@@ -1,10 +1,8 @@
 import CreateAccountPage from 'app/page/create-account-page';
 import GoogleLoginPage from 'app/page/google-login';
-import {getPropValue} from 'utils/element-utils';
-
+import { getPropValue } from 'utils/element-utils';
```
Comment (aweng98) [inline e2e/tests/nightly/registration.spec.ts:3]: file formatted.

File: e2e/tests/notebook/reader-copy-to-workspace.spec.ts [hunk 10]
```diff
@@ -4,19 +4,30 @@ import Link from 'app/element/link';
 import WorkspaceAboutPage from 'app/page/workspace-about-page';
 import WorkspaceAnalysisPage from 'app/page/workspace-analysis-page';
 import WorkspaceDataPage from 'app/page/workspace-data-page';
-import {MenuOption, Language, LinkText, ResourceCard, WorkspaceAccessLevel} from 'app/text-labels';
-import {config} from 'resources/workbench-config';
-import {makeRandomName} from 'utils/str-utils';
-import {createWorkspace, findOrCreateWorkspace, signInWithAccessToken, signInAs, signOut} from 'utils/test-utils';
-import {waitWhileLoading} from 'utils/waits-utils';
+import {
```
Comment (aweng98) [inline e2e/tests/notebook/reader-copy-to-workspace.spec.ts:7]: file formatted.

File: e2e/.eslintrc.js [hunk 11]
```diff
@@ -10,4 +16,80 @@ module.exports = {
...
+    project: './tsconfig.json',
+    ecmaVersion: 2018,
+    sourceType: 'module'
+  },
+  ignorePatterns: ['tsc-out', 'node_modules', '.eslintrc.js'],
+  rules: {
+    // 2 == error, 1 == warning, 0 == off
+    'max-len': [
+      2,
+      140,
```
Comment (aweng98) [inline e2e/.eslintrc.js:29]: I'd like to change it to 120 in future.

**Comment by aweng98** [comment]
@myessail Hi Michael, can you look at this PR for tslint -> eslint migration? It's a large PR with many changed files due to many eslint fixes. But you could shorten the review time by only reviewing few changed files I've mentioned in PR description.

File: e2e/app/component/annotations-sidebar.ts [hunk 12]
```diff
@@ -72,22 +71,22 @@ export default class AnnotationsSidebar extends BaseHelpSidebar {
...
-    const button = await Button.findByName(this.page, {name: LinkText.Edit}, this);
+    const button = Button.findByName(this.page, { name: LinkText.Edit }, this);
     await button.click();
     const modal = new EditDeleteAnnotationsModal(this.page);
     await modal.waitForLoad();
     return modal;
   }
 
   // text area
-  async getAnnotationsTextArea(): Promise<Textarea> {
```
Comment (myessail) [inline e2e/app/component/annotations-sidebar.ts:83]: i'm not 100% familiar with typescript yet but i would prefer if we kept these `async` and `await` keywords unless they really aren't needed for some reason? i looked up the rule that's doing this and if these really aren't async then i'm fine with deleting them

File: e2e/.prettierrc.js [hunk 13]
```diff
@@ -0,0 +1,7 @@
+module.exports = {
+  trailingComma: 'none',
+  singleQuote: true,
```
Comment (myessail) [inline e2e/.prettierrc.js:3]: it looks like this used to be set to `all`, is that still something we want?

File: e2e/app/component/annotations-sidebar.ts (hunk 12 above)
Comment (aweng98) [reply e2e/app/component/annotations-sidebar.ts:83]: Yes, these functions don't have async calls inside, so I was able to remove `await` modifier. This type of error was easy to fix. But affected many files.
Two eslint rules which led the change:
`no-return-await`
`require-await`

File: e2e/.prettierrc.js (hunk 13 above)
Comment (aweng98) [reply e2e/.prettierrc.js:3]: In my opinion, no trailing comma is cleaner. Either way, no functional effect.

**Pull request closed without merging**